src/flyte/_internal/imagebuild/docker_builder.py

```diff
@@ -538,7 +577,7 @@ async def _build_image(self, image: Image, *, push: bool = True, dry_run: bool =
             )
 
             for layer in image._layers:
-                dockerfile = await _process_layer(layer, tmp_path, dockerfile)
+                dockerfile = await _process_layer(layer, tmp_path, dockerfile, image._layers)
```

Comment:
Instead of passing all the layers to `_process_layer` to get the ignore file in the copy handler, can we get the dockerignore path at line 578